components/time/src/zone/zone_name_timestamp.rs

```diff
@@ -103,11 +114,44 @@ impl ZoneNameTimestamp {
     /// assert_eq!(recovered_zoned_date_time.time.subsecond.number(), 0); // always zero
     /// ```
     pub fn from_zoned_date_time_iso(zoned_date_time: ZonedDateTime<Iso, UtcOffset>) -> Self {
-        let qh = zoned_date_time.to_epoch_milliseconds_utc() / 1000 / 60 / 15;
-        let qh_clamped = qh.clamp(Self::far_in_past().0 as i64, Self::far_in_future().0 as i64);
-        // Valid cast as the value is clamped to u32 values.
-        let qh_u32 = qh_clamped as u32;
-        Self(qh_u32)
+        let ms = zoned_date_time.to_epoch_milliseconds_utc();
+        Self(match ms {
+            // Special bit patterns for values that appear in TZDB but are not multiples of
+            // 15 minutes.
+            63593070000 => 0xFFFFF0,
+            307622400000 => 0xFFFFF1,
+            576041460000 => 0xFFFFF2,
+            576043260000 => 0xFFFFF3,
+            594180060000 => 0xFFFFF4,
+            607491060000 => 0xFFFFF5,
+            1601740860000 => 0xFFFFF6,
+            1633190460000 => 0xFFFFF7,
+            1664640060000 => 0xFFFFF8,
```

Comment:
Most of these appear to be transitions that occur at xx:01 instead of xx:00. Is that a distinction that is important to support? I don't like how this list is basically unbounded growth; it increases maintenance cost.

Reply:
Please make an issue for 2.1 to bikeshed this.

Reply:
it would need to be the next higher one, not the nearest: if a period starts at 0:40 and we encode that as 0:30, 0:35 will also encode as 0:30 and will match when it shouldn't. if we encode it as 0:45, only 0:40-0:59 will encode the same.

Reply:
I'm saying that if the transition is :31, then we take the :45 that happens after it and special case it to resolve to :31 just that one time

Reply:
ok fine. i did this, anything keeping you from approving?